src/blockchain_db/blockchain_db.h

```diff
@@ -1660,7 +1660,24 @@ class BlockchainDB
   /**
    * @brief fix up anything that may be wrong due to past bugs
    */
-  virtual void fixup();
+  enum struct fixup_type
+  {
+    standard,
+    calculate_difficulty,
+  };
+
+  struct fixup_context
+  {
+    fixup_type type;
+    union
+    {
+      struct
+      {
+        uint64_t start_height;
+      } calculate_difficulty;
```

Comment:
consider renaming to calculate_difficulty_params here to disambiguate with the enum when the two are near one another in the code.

Reply:
Fixed